doc/source/cluster/slurm.rst

```diff
@@ -3,7 +3,188 @@
+
+Note that this is an optional step, but it is often required for enabling the right set of dependencies.
+
+.. code-block:: bash
+
+  # Example: module load pytorch/v1.4.0-gpu
+  # Example: conda activate my-env
+
+  conda activate my-env
+
+Obtain the head IP address
+~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Next, we'll want to obtain a hostname and a node IP address for the head node. This way, when we start worker nodes, we'll be able to properly connect to the right head node.
+
+.. literalinclude:: /cluster/examples/slurm-basic.sh
+   :language: bash
+   :start-after: __doc_head_address_start__
+   :end-before: __doc_head_address_end__
+
+
+
+Starting the Ray head node
+~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+After detecting the head node hostname and head node IP, we'll want to create
+a Ray head node runtime. We'll do this by using ``srun`` as a background task
+as a single task/node (recall that ``tasks-per-node=1``).
+
+Below, you'll see that we explicitly specify the number of CPUs (``num-cpus``)
+and number of GPUs (``num-gpus``) to Ray, as this will prevent Ray from using
+more resources than allocated. We also need to explictly
+indicate the ``node-ip-address`` for the Ray head runtime:
+
+.. literalinclude:: /cluster/examples/slurm-basic.sh
+   :language: bash
+   :start-after: __doc_head_ray_start__
+   :end-before: __doc_head_ray_end__
+
+By backgrounding the above srun task, we can proceed to start the Ray worker runtimes.
+
+Starting the Ray worker nodes
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Below, we do the same thing, but for each worker. Make sure the Ray head and Ray worker processes are not started on the same node.
+
+.. literalinclude:: /cluster/examples/slurm-basic.sh
+   :language: bash
+   :start-after: __doc_worker_ray_start__
+   :end-before: __doc_worker_ray_end__
+
+Submitting your script
+~~~~~~~~~~~~~~~~~~~~~~
+
+Finally, you can invoke your Python script:
+
+.. literalinclude:: /cluster/examples/slurm-basic.sh
+   :language: bash
+   :start-after: __doc_script_start__
+
+
+Python-interface SLURM scripts
+------------------------------
+
+[Contributed by @pengzhenghao] Below, we provide a helper utility (:ref:`slurm-launch.py <slurm-launch>`) to auto-generate SLURM scripts and launch.
+``slurm-launch.py`` uses an underlying template (:ref:`slurm-template.sh <slurm-template>`) and fills out placeholders given user input.
+
+You can feel free to copy both files into your cluster for use. Feel free to also open any PRs for contributions to improve this script!
+
+Usage example
+~~~~~~~~~~~~~
+
+If you want to utilize a multi-node cluster in slurm:
+
+.. code-block:: bash
+
+    python slurm-launch.py --exp-name test --command "python your_file.py" --num-nodes 3
+
+If you want to specify the computing nodes, just use the same node name in ``sinfo`` command:
+
+.. code-block:: bash
+
+    python slurm-launch.py --exp-name test --command "python your_file.py" --num-nodes 3 --node chpc-cn[003-005]
```

Comment:
Maybe change the name `chic-cn[003-005]` to a placeholder like `NODE_NAMES` or so would be better.